WordPress/gutenberg
# [Mobile] - Prevent passing potential `false` values to the `onPress` prop

## What?
This PR aims to fix a potential crash where it appears an `onPress` prop has a `false` value. 

## Why?
There are a few crash events on low-end devices where the editor crashes in the [Pressability](https://github.com/wordpress-mobile/gutenberg-mobile/issues/6359) component of React Native. After several different investigations, we haven't found reproducible steps of this problem.

## How?
This PR prevents passing optional values to the `onPress` prop of the `ButtonBlockAppender` component and `MediaEdit` button in the Cover block.

With these changes, a function will always be passed with checks to prevent passing potential `false` values.

## Testing Instructions
> [!NOTE]  
> It's recommended to smoke test the following as there are no reproducible steps

### ButtonBlockAppender - Test case 1

- Open the app with metro running
- Tap on the `+` button to add a block
- Expect the block to be added

### ButtonBlockAppender - Test case 2

- Open the app with metro running
- Tap on the `+` button to add a Columns block
- Select the first Column
- Tap on the inline `+` appender button and add a block
- Expect the block to be added

### ButtonBlockAppender - Test case 3

- Open the app with metro running
- Tap on the `+` button to add a Buttons block
- Tap on the inline `+` appender button and add another Button block
- Expect the block to be added

### Cover Block MediaEdit

- Open the app with metro running
- Tap on the `+` button and add a Cover block
- Tap a color from the inline color palette of the block
- Select the parent Cover block
- Tap on the top-right MediaEdit button
- **Expect** the picker to show up

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

## ButtonBlockAppender

https://github.com/WordPress/gutenberg/assets/4885740/51184f2d-ad7f-48dc-98f7-d6879cb54f0d

## Cover Block MediaEdit

https://github.com/WordPress/gutenberg/assets/4885740/9802588a-7983-453a-8eff-3dd2a130b449
